### ISSUE TITLE
Support for nested tags

### ISSUE THREAD

**Issue by tophee** [HeroBlackInk/ultimate-todoist-sync-for-obsidian#50]
Not sure if this is a bug report or a feature request but I noticed that tags don't sync properly for me and I think that has to do with the fact that I mostöy use nested tags. For example, all my tags designating a project start with `#p/` so that the Garden project would be tagged `#p/garden`, but the plugin only syncs the top level tag (#p) to todoist, not the rest.

So, what I'm saying is: it would be nice if the entire tag could be synced (including the "path")

**Comment by torstenfeld** [comment]
I'd love to see this change as well. I use the tag paths a lot for, e.g., #people/xyz or #project/xyz.

**Comment by torstenfeld** [comment]
I created a pull request which should fix the issue.

**Comment by tophee** [comment]
Haha, one extra character is all it took!

However, will this work with tags that don’t end with a `\`?

**Comment by torstenfeld** [comment]
Yes, the position of the / doesn't matter. If you want to understand the regex better, I recommend to go to https://regex101.com/ and enter the regex into the input field at the top. Choose "ECMAScript" as the flavor on the left hand side. Afterwards, the regex will be explained in the right hand side of the page. 

The regex to enter:
`(^|\s)(#[a-zA-Z\d\u4e00-\u9fa5-/]+)`

**Comment by torstenfeld** [comment]
And in case you do not want to wait for a new release, you can go to the plugins directory of Obsidian and make the changes manually in the main.ts file of the plugin.

**Comment by tophee** [comment]
> Yes, the position of the / doesn't matter.

of course not. That was a stupid comment by me. I should have looked more carefully at the regex in the PR, then I would have seen the square brackets… Now when you paste it here it is obvious. 

Sorry about that. And thanks for the fix.

BTW: what would it take to replace underscores with Spaces (which are allowed in Todoist but not in Obsidian)?
